packages/web/src/components/upload/UploadChip.tsx

```diff
@@ -30,58 +51,112 @@ type UploadChipProps = {
   /**
    * Is this upload the user's first of this type
    * */
+  source: 'nav' | 'profile' | CreatePlaylistSource
   isFirst?: boolean
-  onClick: () => void
-  isArtist?: boolean
 }
 
 const UploadChip = ({
   type = 'track',
   variant = 'tile',
-  isArtist = false,
   isFirst = false,
-  onClick
+  source
 }: UploadChipProps) => {
+  const { isEnabled: isEditAlbumsEnabled } = useFlag(FeatureFlags.EDIT_ALBUMS)
+  const messages = getMessages(type)
   const icon =
-    type === 'track' || type === 'album' ? (
+    type === 'track' ? (
       <IconUpload className={styles.iconUpload} />
     ) : (
-      <IconMultiselectAdd className={styles.iconPlus} />
+      <IconPlus className={styles.iconPlus} color='subdued' />
     )
 
-  let text
+  let text: string
   switch (type) {
     case 'track':
       text = variant === 'nav' ? messages.track : messages.aTrack
       break
     case 'album':
-      text = isFirst ? messages.firstAlbum : messages.album
-      break
     case 'playlist':
-      if (isArtist) {
-        text = isFirst ? messages.firstArtistPlaylist : messages.artistPlaylist
-      } else {
-        text = isFirst ? messages.firstPlaylist : messages.playlist
-      }
+      text = messages.createNewCollection(isFirst)
       break
     default:
       break
   }
 
-  return (
+  const dispatch = useDispatch()
+
+  const handleClickUpload = useCallback(() => {
+    dispatch(pushRoute(UPLOAD_PAGE))
```

Comment:
would it be too hard to make this react-router links? on top of that it would be excellent to have links that can take "analytics" objects to call on click, but maybe thats too much for now

Reply:
Oh, yeah that's a good idea. As you mentioned below, maybe a little tough here, but something we should support.